blog/2020/08/invalidations.md

```diff
@@ -0,0 +1,367 @@
+invalidation is sometimes necessary if you want both good runtime performance and interactive usage, and this combination is one of the best things about Julia.
+The real question is whether there are *unnecessary* invalidations,
+or an unexploited strategy to limit their impact.
+Determining the answer to that question requires that we develop an understanding the common sources of invalidation and what can be done to fix them.
+
+## Changes in Julia that have reduced the number of invalidations
+
+Much of the progress in reducing invalidations in Julia 1.6 come from changes to generic mechanisms in the compiler.
+A [crucial change](https://github.com/JuliaLang/julia/pull/36733) was the realization that some invalidations were formerly triggered by new methods of *ambiguous specificity* for particular argument combinations; since calling a function with this argument combination would have resulted in a `MethodError` anyway, it is unnecessary to invalidate their (nominal) dependents.
+
+Another [pair](https://github.com/JuliaLang/julia/pull/36208) of [fundamental](https://github.com/JuliaLang/julia/pull/35904) changes led to subtle but significant alterations in how much Julia specializes poorly-inferred code to reflect the current state of the world.
+By making Julia less eager to specialize in such circumstances, huge swaths of the package ecosystem became more robust against invalidation.
+
+Most of the remaining improvements stem from more directed changes that improved inferrability of dozens of specific methods.  You find many examples by searching for the [latency tag](https://github.com/JuliaLang/julia/pulls?q=is%3Apr+label%3Alatency+is%3Aclosed) among closed pull requests.
+
+Most of these changes were at least partially inspired by new tools to analyze the source of invalidations, and we briefly touch on this final topic below.
+
+## Tools for analyzing and fixing invalidations
+
+Recently, the [SnoopCompile] package gained the ability to analyze invalidations and help developers fix them.
+Because these tools will likely change over time, this blog post will only touch the surface; people who want to help fix invalidations are encouraged to read [SnoopCompile's documentation](https://timholy.github.io/SnoopCompile.jl/stable/snoopr/) for further detail.
+There is also a [video](https://www.youtube.com/watch?v=7VbXbI6OmYo) available with a live-session fixing a real-world invalidation, which might serve as a useful example.
+
+But to give you a taste of what this looks like, here are a couple of screenshots.
+These were taken in Julia's REPL, but you can also use these tools in [vscode] or other environments.
+
+First, let's look at a simple way (one that is not always precisely accurate, see SnoopCompile's documentation) to collect data on a package (in this case, loading the SIMD package):
+
+![snoopr_simd](/assets/blog/2020-invalidations/SIMD_invalidations.png)
```

Comment:
Not sure if this is my case only. The words in this screenshot are of smaller fonts than others.

![image](https://user-images.githubusercontent.com/8684355/90343885-34b81300-e047-11ea-964c-830f5937426f.png)

Reply:
Yeah, the output is fairly wide for the first one and it squashes to the width of the page. We can either choose to truncate and keep the fontsize similar or see the full display. Preference?

Reply:
I prefer to keep it as it is if reproducing a similar screenshot is non-trivial since you probably already fixed that :P